### ISSUE TITLE
cider-jack-in doesn't work well with boot

### ISSUE THREAD

**Issue by lispyclouds** [clojure-emacs/cider#2644]
## Expected behavior
After cider-jack-in, I should be able to use CIDER and navigate normally.

## Actual behavior
cider-jack-in succeeds but causes emacs to freeze for a couple of seconds and the following error is seen:
![Screen Shot 2019-06-04 at 9 47 30 AM](https://user-images.githubusercontent.com/5615588/58866582-2a675e80-86b9-11e9-9d12-f0c8dd50da28.png)
Other issues:
- jump-to-definition also doesn't work after this.
- The cider repl is in an uninitialized state
![Screen Shot 2019-06-04 at 11 33 26 AM](https://user-images.githubusercontent.com/5615588/58868422-a0b99000-86bc-11e9-92fb-8113242a8eea.png)

## Steps to reproduce the problem
- Use any clojure project using boot
- Perform `cider-jack-in`
- Observe emacs freeze and the error appear

### CIDER version information

```
CIDER master
Clojure 1.10
Java 12
```

### Lein/Boot version

Boot 2.8.3

### Emacs version

Emacs 26.2 with Spacemacs develop

### Operating system

MacOS 10.14.5

**Comment by lispyclouds** [comment]
Possibly related to https://github.com/clojure-emacs/cider/pull/2643

**Comment by bbatsov** [comment]
@arichiardi @alexander-yakushev Can you confirm the problem?

**Comment by alexander-yakushev** [comment]
It works fine for me on Java 8 and Boot 2.8.2.

Could you please downgrade to Boot 2.8.2 for a test? That newer build of Boot doesn't seem to work correctly in many regards.

**Comment by bbatsov** [comment]
It doesn't? I thought there were virtually no internal changes there.

**Comment by lispyclouds** [comment]
2.8.2 on Java 12 has the same issues as well

**Comment by lispyclouds** [comment]
Tried running boot 2.8.2 on java 1.8.0_212 with no luck as well.

**Comment by arichiardi** [comment]
I tried only `2.8.3` and it seemed to be working

**Comment by arichiardi** [comment]
I noticed that the nRepl launcher does some `ask` of dancing so that might be the difference

**Comment by alexander-yakushev** [comment]
Is there anything in your `profile.boot` that can influence this?

**Comment by lispyclouds** [comment]
@arichiardi you're right it does seem to work on a sample project i tried: https://github.com/oakes/full-stack-boot-example but i can see the issue on mine for some reason: https://github.com/bob-cd/bob

**Comment by lispyclouds** [comment]
@alexander-yakushev I don't really have a profile.boot. just the `boot.properties` file

**Comment by arichiardi** [comment]
When I get to the office I am going to try your project, thanks and sorry. I also had one other version using `nrepl` directly so @bbatsov might want to revert to that instead.

**Comment by lispyclouds** [comment]
Thanks a lot!

**Comment by alexander-yakushev** [comment]
@lispyclouds Actually, if you are trying to jack in with `cider-nrepl 0.22.0-beta2`, it is broken for me as well.

**Comment by bbatsov** [comment]
> When I get to the office I am going to try your project, thanks and sorry. I also had one other version using nrepl directly so @bbatsov might want to revert to that instead.

Let's figure out what's wrong first. We've used Boot's default task for a very long time, so I'm really puzzled by this development. 

> I noticed that the nRepl launcher does some ask of dancing so that might be the difference

I guess you mean ack, not ask, right? That's something that Lein does as well, and it's pretty simple. Not sure how many clients (if any) rely on ack, but I doubt it's causing the problem, as it has always been there.

**Comment by lispyclouds** [comment]
@alexander-yakushev yes that's what i was trying with

**Comment by arichiardi** [comment]
> I guess you mean ack, not ask, right?

Yes sorry was on the phone

**Comment by arichiardi** [comment]
Can confirm the stuck emacs and the timeout in `bob`.:

```
error in process filter: nrepl-send-sync-request: Sync nREPL request timed out (op describe)
error in process filter: Sync nREPL request timed out (op describe)
```

**Comment by bbatsov** [comment]
Any idea what's the real reason? That error message simply means that nREPL didn't boot up properly. What's happens when you run the boot shell command outside of Emacs?

**Comment by arichiardi** [comment]
Will try that as well

**Comment by bbatsov** [comment]
@arichiardi Any progress with the investigation?

**Comment by arichiardi** [comment]
Didn't have time, will try this week hopefully

**Comment by lispyclouds** [comment]
Hello, is there any updates on this? I poked around a bit and it really seems only my source code produces this issue. If there is some way I can help I'm happy to do it! 😄 I have a feeling this could be some issue in my build.boot?

**Comment by bbatsov** [comment]
@lispyclouds I'm still waiting for someone to report that's the actual problem (which should be reproducible by running the command CIDER generated in some terminal).

**Comment by arichiardi** [comment]
Sorry been travelling and kind of on holiday at the moment, not sure I can carve some time for this but will try to

**Comment by lispyclouds** [comment]
This has been solved now. As I suspected, the issue was in my `build.boot` evaling the main ns which took a while for me as I had a mount state there. That caused CIDER to timeout while waiting for nREPL to come up. Setting  something like `(setq nrepl-sync-request-timeout 100)` fixed this! Found the fix when hacking with @bbatsov himself and can't be more thankful to him and the entire CIDER team! 🙏
